### ISSUE TITLE
Scrubber needs improvement

### ISSUE THREAD

**Issue by nziebart** [palantir/atlasdb#2408]
There are a few issues with background scrub that make it unusable in its current state:

1. `TransactionManagers.create` does not start the scrubber
2. `Scrubber#start` does not actually run a continuous background job. It just schedules one run on an executor and then exits (unclear if this is a bug or intentional; seems like a bug)
3. `Scrubber` does not acquire the backup lock when deleting. We are trying to move away from the backup lock, but at the moment it's required to ensure backups are consistent.
4. `Scrubber` is inefficient and likely duplicates work, since it does not take a lock like sweep does
5. `Scrubber` is woefully untested, and does a lot of the same things sweep does without sharing any code

**Comment by tpetracca** [comment]
> `Scrubber` does not acquire the backup lock when deleting. We are trying to move away from the backup lock, but at the moment it's required to ensure backups are consistent.

I wonder if we care. Generally the scrubber is used for hard delete in places where we are contractually obligated to delete data within an amount of time. Might be worth talking to a customer that actually has to do this to understand how they think about their backups when this occurs (because in theory that data will still exist in their backups from before that point in time). This is also why the scrubber violates the transaction protocol (and why locking the immutable TS in place of the persisted backup lock will not work here)

**Comment by nziebart** [comment]
I would agree with this for the case of `AGGRESSIVE_HARD_DELETE` (since that mode will block your transaction until the data is deleted). But for regular `HARD_DELETE`, cells just go into a queue to be swept asynchronously. In this mode, we still allow the usual grace period of 1 hour for the unreadable timestamp, so I feel like it makes sense to mirror sweep in making sure backups are consistent.

The motivation here is that internal navigation product is considering using scrub to ensure quick removal of tombstones for range scan perf.

**Comment by j-baker** [comment]
Additionally, the scrubber will probably cause Cassandra OOMs on current internally deployed Cassandras if we do turn it on, since it will range scan over Cassandra deletion sentinels.
